### PR TITLE
[AIRFLOW-3447] Add 2 options for ts_nodash Macro

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3447

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Add Option to use `ts_nodash` and `ts_nodash_with_tz` macro so it gives any option to users to use it Filename. Previous users have been using `ts_nodash` for file and folder names.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Added tests

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
Updated the docs

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Review by ashb — COMMENTED** [review]
Let's add this to UPDATING - while the change is good, it might change a few things for some people.

**Comment by ashb** [comment]
Actually - what does this do to the filename that logs are written to?

**Comment by kaxil** [comment]
I don't think it affects the log filename as the log filename uses `ts`. I will add details to UPDATING.md

```
# Log filename format
log_filename_template = {{{{ ti.dag_id }}}}/{{{{ ti.task_id }}}}/{{{{ ts }}}}/{{{{ try_number }}}}.log
log_processor_filename_template = {{{{ filename }}}}.log
dag_processor_manager_log_location = {AIRFLOW_HOME}/logs/dag_processor_manager/dag_processor_manager.log
```

File: UPDATING.md [hunk 1]
```diff
@@ -24,6 +24,13 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Modification to `ts_nodash` macro
+`ts_nodash` previously contained TimeZone information alongwith execution date. For Example: `20150101T000000+0000`. This is not user-friendly for file or folder names which was a popular use case for `ts_nodash`. Hence this behavior has been changed and using `ts_nodash` will no longer contain TimeZone information. And a new macro `ts_nodash_with_tz` has been added which can be used to get a string with execution date and timezone info without dashes.
```
Comment (ashb) [inline UPDATING.md]: Something to say "this restores the pre-1.10 behaviour of this macro"?

Comment (kaxil) [reply UPDATING.md]: Done.

**Review by ashb — APPROVED** [review]

**Pull request merged**